src/components/NewCommentForm.tsx

```diff
@@ -1,99 +1,119 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
+import { addComment } from '../api';
+import { GlobalContext } from '../reducer';
+import { Comment } from '../types/Comment';
+import { Error } from '../types/Error';
+import { Field } from './Field';
 
-export const NewCommentForm: React.FC = () => {
-  return (
-    <form data-cy="NewCommentForm">
-      <div className="field" data-cy="NameField">
-        <label className="label" htmlFor="comment-author-name">
-          Author Name
-        </label>
+type Props = {
+  getComments: (id:number) => void
+};
 
-        <div className="control has-icons-left has-icons-right">
-          <input
-            type="text"
-            name="name"
-            id="comment-author-name"
-            placeholder="Name Surname"
-            className="input is-danger"
-          />
+export const NewCommentForm: React.FC<Props> = ({ getComments }) => {
+  const [state, dispatch] = useContext(GlobalContext);
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [text, setText] = useState('');
```

Comment:
it could be one state with object

Reply:
how i can change one field object if i don't know other value fields, i can get it's every Field but it will be more and hard code

Reply:
example: 

```
  const handleChange = (e: React....) => {
  setFields({...fields, [e.target.name]:e.target.value});
}

```
but you need to pass for every input `name` property